modules/calib3d/include/opencv2/calib3d.hpp

```diff
@@ -1614,6 +1615,58 @@ CV_EXPORTS_W double calibrateCamera( InputArrayOfArrays objectPoints,
+#calibrateCamera for details.
+@param imageSize Size of the image used only to initialize the intrinsic camera matrix.
+@param cameraMatrix Output 3x3 floating-point camera matrix. See #calibrateCamera for details.
+@param distCoeffs Output vector of distortion coefficients. See #calibrateCamera for details.
+@param rvecs Input-output vector of rotation vectors estimated for each pattern view. If @ref
+CALIB_FIX_EXTRINSIC is specified, rotation matrices and translation vectors equal in size to
+the number of images must be initialized and will be used for calibration.  See #calibrateCamera
+for details.
+@param tvecs Input-output vector of translation vectors estimated for each pattern view.
+@param stdDeviationsIntrinsics Output vector of standard deviations estimated for intrinsic parameters.
+See #calibrateCamera for details.
+@param stdDeviationsExtrinsics Output vector of standard deviations estimated for extrinsic parameters.
+See #calibrateCamera for details.
+parameter is ignored with standard calibration method.
+ @param perViewErrors Output vector of the RMS re-projection error estimated for each pattern view.
+@param flags Different flags that may be zero or a combination of some predefined values. See
+#calibrateCamera for details.
+@param criteria Termination criteria for the iterative optimization algorithm.
+
+-   @ref CALIB_FIX_EXTRINSIC The input rotation matrices and translation vectors are used for
+calibration and are not changed during optimization.
+
+@return the overall RMS re-projection error.
+ */
+CV_EXPORTS_AS(calibrateCameraWithExtrinsicsExtended) double calibrateCameraWithExtrinsics( InputArrayOfArrays objectPoints,
+                                     InputArrayOfArrays imagePoints, Size imageSize,
+                                     InputOutputArray cameraMatrix, InputOutputArray distCoeffs,
+                                     InputOutputArrayOfArrays rvecs, InputOutputArrayOfArrays tvecs,
+                                     OutputArray stdDeviationsIntrinsics,
+                                     OutputArray stdDeviationsExtrinsics,
+                                     OutputArray perViewErrors,
+                                     int flags = 0, TermCriteria criteria = TermCriteria(
+                                        TermCriteria::COUNT + TermCriteria::EPS, 30, DBL_EPSILON) );
+
+/** @overload */
+CV_EXPORTS_W double calibrateCameraWithExtrinsics( InputArrayOfArrays _objectPoints,
+                                     InputArrayOfArrays imagePoints, Size imageSize,
+                                     InputOutputArray cameraMatrix, InputOutputArray distCoeffs,
+                                     InputOutputArrayOfArrays rvecs, InputOutputArrayOfArrays tvecs,
+                                     int flags = 0, TermCriteria criteria = TermCriteria(
+                                        TermCriteria::COUNT + TermCriteria::EPS, 30, DBL_EPSILON) );
```

Comment:
@alalek @asmorkalov I propose a dedicated api to support passing in extrinsics.  Can you see if you're agreeable with the api?

The extrinsics type is `InputOutputArrayOfArrays`, which allows OpenCV to modify it in case we want to support an initial guess for the extrinsics in the future.